### PR TITLE
Some fixes to getBBox and much more

### PR DESCRIPTION
**Pull request by apla**
Hi,

First of all, thank you so much for this project. Finally I've got server-side SVG rendering for my project.

Unfortunately I've spend almost a week to make things works. Commits in this repository contains tests and fix for the found issues, I don't want to split those commits into separate pull requests.

### CODE REVIEW

File: utils/bboxUtils.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const bbox = (node, applyTransformations) => {
         parseFloat(node.getAttribute('y')) || 0,
         parseFloat(node.getAttribute('width')) || 0,
         parseFloat(node.getAttribute('height')) || 0
-      )
+      ).transform(node.matrixify());
     case 'svg':
```
Comment (Fuzzyma) [inline utils/bboxUtils.js]: When you do that, you transform the bbox with the transformation matrix of the element. This is not how bbox is defined. Bbox always returns the bounding box in userspace. That means its the coordinates in the transformed space. If you apply the transformation you basically say: I want the bbox in space of the parent element.

Or am I totally wrong here?

**Comment by Fuzzyma** [comment]
Wow thats so much. Thank you! Also that tests are nice. I used to check if all sv.js test runs with svgdom (yes iam lazy - sorry!).

I added a comment which you might want to answer. Despite from this it looks pretty good

File: utils/bboxUtils.js (hunk 1 above)
Comment (apla) [reply utils/bboxUtils.js]: I'll check with chromium rendering. I have (unpublished) [puppeteer](https://github.com/GoogleChrome/puppeteer) test and I have to convert test to run on node and chromium and compare results.

Comment (apla) [reply utils/bboxUtils.js]: Agree, my fault. Fixed with adbd079

**Comment by apla** [comment]
Now all the tests pass in the browser and in node (you can even check visually). I have to add `puppeteer` devDependency in package.json, but `puppeteer` team decided to include full chromium browser on `npm install` and I not like this behaviour.

So, to launch svgdom tests in node environment, just run `npm test`. To launch same assertions and svg modifications in the chromium browser, install puppeteer with `npm i -D puppeteer` and launch test like `TEST_BROWSER=1 npm test`

**Comment by apla** [comment]
Hi, can you check `svg.js` with those changes?

**Comment by Fuzzyma** [comment]
I will look into this the next month. Sorry for the delay. Quite busy at the moment

**Pull request merged**